### ISSUE TITLE
Inconsistent usage of class keys for similar actions on resources

### ISSUE THREAD

**Issue by matdave** [modxcms/revolution#13733]
### Summary
When viewing the manager log, it appears that there is some inconsistency when using class keys for certain actions.  Sometimes it falls on the overall modResource class, and other times it uses the specific class, e.g. modDocument or modWeblink.

### Step to reproduce
Publish and Unpublish a resource, then check the manager actions log. 

### Observed behavior
See how those two actions are logged.  Publishing uses the modDocument class key, and unpublishing uses the modResource class key.

### Expected behavior
It would be nice if all actions specific to a resource used the modResource class.

### Environment
MODX 2.6.1

**Comment by OptimusCrime** [comment]
I could be wrong, but I think this is due to inheritance and where the method is called. If publishing is doing within the `modDocument` class, and it calls something like `get_class($this)` it will say `modDocument`. However, if the method for unpublishing is so similar, that it can be in the (shared) parent class `modResource` then the log entry `get_class($this)` will say `modResource`. 

Due to the way inheritance and OOP work, there is no easy way to solve this without yanking out the method from the shared parent class into the individual classes instead.

Have you looked at the code and identified the cause?

**Comment by matdave** [comment]
I have not had time to really dive into this. I guess I could retract my previous statement and say that it would make more sense if everything used the object class that it was defined as, as all actions of the parent class would be available with the addition of any extended processes on the defined class. 

I'll see if I can find some more time over the next couple weeks to document why the parent class gets used the majority of the time, while the extended class only shows up on occasion.

**Comment by OptimusCrime** [comment]
Yeah, my gut is that some of the code is so general it can be used across all inherited classes, while other things are implemented specific.
